city/crud.py

```diff
@@ -0,0 +1,62 @@
+from fastapi import HTTPException
+
+from sqlalchemy import select, insert, delete
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from city import models, schemas
+
+
+async def get_all_cities(db: AsyncSession) -> list[models.City]:
+    query = select(models.City)
+    cities_list = await db.execute(query)
+
+    return [city[0] for city in cities_list.fetchall()]
+
+
+async def create_city(db: AsyncSession, city: schemas.CityCreate) -> dict:
+    query = insert(models.City).values(
+        name=city.name,
+        additional_info=city.additional_info,
+    )
+    created_city = await db.execute(query)
+    await db.commit()
+    response = {**city.model_dump(), "id": created_city.lastrowid}
+
+    return response
+
+
+async def retrieve_city(db: AsyncSession, id: int) -> models.City:
+    city = await db.get(models.City, id)
```

Comment:
The return message in 'update_temperatures' function has an extra space at the beginning of the string. Consider removing it for consistency.